### ISSUE TITLE
Default term type for function value in object map

### ISSUE THREAD

**Issue by pmaria** [kg-construct/rml-fno-spec#5]
The spec says that 
> [the default term type depends on whether the term map is an object map or not](https://www.w3.org/TR/r2rml/#termtype).

which points to the R2RML spec which says
>If the term map does not have a `rr:termType` property, then its term type is:
>* rr:Literal, if it is an object map and at least one of the following conditions is true:
>   * It is a column-based term map.
>   * It has a `rr:language` property (and thus a specified language tag).
>   * It has a `rr:datatype` property (and thus a specified datatype).
>* rr:IRI, otherwise.

Reading this I would say an `fnml:FunctionMap` with no `rr:termType`,  ` rr:language` or `rr:datatype` specified should by default return an `rr:IRI`. (Note: this is how it is implemented in CARML currently)

In FnO test case [RMLFNOTC0009-CSV](https://github.com/RMLio/rml-fno-test-cases/tree/master/RMLFNOTC0009-CSV) however, the expected term type in such a case is `rr:Literal`. We need to clarify this.

**Comment by bjdmeest** [comment]
Ah yes, and right now [the default term type depends on whether the term map is an object map or not](https://www.w3.org/TR/r2rml/#termtype) is (implicitly) extended into

>If the term map does not have a `rr:termType` property, then its term type is:
>* rr:Literal, if it is an object map and at least one of the following conditions is true:
>   * It is a column-based term map **or a also fnml:FunctionMap**
>   * It has a `rr:language` property (and thus a specified language tag).
>   * It has a `rr:datatype` property (and thus a specified datatype).
>* rr:IRI, otherwise.

I have no real opinion for what should be the default case. I'll clarify in the spec what the current situation is, but I don't want to force that as solution. Maybe we should just vote or smth?

**Comment by andimou** [comment]
detail but in the case of RML (given that FnO is aligned with RML but not R2RML),
it should be **reference-based term map** (instead of column-based) or function map 

before reaching the point of voting, should we perhaps look into our use cases or call for use cases?

The use cases might give us insights regarding what might be the most commonly intuitive.

**Comment by bjdmeest** [comment]
> detail but in the case of RML (given that FnO is aligned with RML but not R2RML),
> it should be **reference-based term map** (instead of column-based) or function map

You're right and I also had that insight before changing the spec, it's now as below, in accordance with the [current RML spec](https://rml.io/specs/rml/v/1.1.0/)

>If the term map does not have a `rr:termType` property, then its [term type](https://rml.io/specs/rml/#term-type) is:
>* `rr:Literal`, if it is an [object map](https://www.w3.org/TR/r2rml/#dfn-object-map) and at least one of the following conditions is true:
>   * It is a [reference-based term map](https://rml.io/specs/rml/#reference-valued-term-map),  **or also a fnml:FunctionMap**
 >  * It has a `rml:languageMap` and/or `rr:language` property (and thus a [language map](https://rml.io/specs/rml/#language-map) and/or a [specified language tag](https://rml.io/specs/rml/#specified-language-tag)).
>   * It has a `rr:datatype` property (and thus a [specified datatype](https://rml.io/specs/rml/#specified-datatype)).
>* `rr:IRI`, otherwise.

:+1: on the use cases :)

**Comment by andimou** [comment]
>  It is a reference-based term map, or also a fnml:FunctionMap

getting picky with details now but the one is in a human readable format, ie reference-based term map and the other in machine-readable format, ie `fnml:FunctionMap`.

Better keep both human-readable: reference-based term map and function map

or both in machine-readable format: `rr:TermMap` with `rml:reference` and `fnml:FunctionMap`

or a hybrid :)

(I'm personally in favor of the former)

**Comment by bjdmeest** [comment]
You're right, added definition a "function map" and used that one where relevant, see change at 506737a2c9f952d9aebcb651176f8e00dc801057

**Comment by andimou** [comment]
similar question here, do we have use cases in the end for this? If not, maybe we need to consider having a collective call for issues that we want to work on and we don't have use cases?

**Comment by samiscoding** [comment]
Considering this definition (and for now `fnml:FunctionMap` as a term map), if the `FunctionMap` is a `rr:subjectMap` and no `rr:termType` is provided then the default type is `rr:IRI`, did I understand it correctly? 

> Ah yes, and right now [the default term type depends on whether the term map is an object map or not](https://www.w3.org/TR/r2rml/#termtype) is (implicitly) extended into
> 
> > If the term map does not have a `rr:termType` property, then its term type is:
> > 
> > * rr:Literal, if it is an object map and at least one of the following conditions is true:
> >   
> >   * It is a column-based term map **or a also fnml:FunctionMap**
> >   * It has a `rr:language` property (and thus a specified language tag).
> >   * It has a `rr:datatype` property (and thus a specified datatype).
> > * rr:IRI, otherwise.
> 
> I have no real opinion for what should be the default case. I'll clarify in the spec what the current situation is, but I don't want to force that as solution. Maybe we should just vote or smth?
